src/db.ts

```diff
@@ -29,27 +28,63 @@ export const init: () => void = () => {
       namedPlaceholders: true,
     });
 
-    // TODO: set timezone
-
-    // console.debug('MySql Adapter Pool generated successfully');
+    // TODO: set timezone?
   } catch (error) {
-    // console.error('[mysql.connector][init][Error]: ', error);
-    throw new DatabaseError('Failed to initialized pool');
+    throw new DatabaseError('Failed to initialized pool', undefined, error);
   }
-};
+}
 
-export const execute = <T>(query: string, params: string[] | Object): Promise<T> => {
+export async function execute<T>(query: string, params: string[] | Object): Promise<T> {
   if (null === pool) throw new DatabaseError('Pool was not created. Ensure pool is created when running the app.');
 
   try {
-    return new Promise<T>((resolve, reject) => {
-      pool!.query(query, params, (error, results) => {
-        if (error) reject(error);
-        else resolve(results as unknown as T);
-      });
-    });
+    const [rows] = await pool.query<RowDataPacket[]>(query, params);
 
+    return rows as unknown as T;
   } catch (error) {
     throw new DatabaseError('Failed to execute MySQL query', query, error);
   }
-};
+}
+
+export async function executeInConnection<T>(connection: PoolConnection, query: string, params: string[] | Object): Promise<T> {
+  try {
+    const [rows] = await connection.query<RowDataPacket[]>(query, params);
+
+    return rows as unknown as T;
+  } catch (error) {
+    throw new DatabaseError('Failed to execute MySQL query', query, error);
+  }
+}
+
+export async function querySingleResult<T>(query: string, params: string[] | Object): Promise<T|null> {
+  if (-1 === query.indexOf('LIMIT')) {
+    query = `${query} LIMIT 1`;
+  }
+  const results = await execute<T[]>(query, params);
+
+  return results.length ? results[0] : null;
+}
+
+export async function querySingleScalarResult<T>(query: string, params: string[] | Object): Promise<T|null> {
+  if (-1 === query.indexOf('LIMIT')) {
+    query = `${query} LIMIT 1`;
+  }
```

Comment:
if you reuse `querySingleResult`, I don't see the point of rechecking the limit thing here as well.